### DISCUSSION TITLE
[例]フォーム入力値を関数で受け取る場合引数をどのように設定するか

### DISCUSSION THREAD

**Discussion by fukaeridesui** [fukaeridesui/suipstakes#3] [Ideas]
抽選イベントを作成するフォームに入力した直をコントラクトの関数に渡す場合を考える
引数の受け方としてどのような方法が考えられるか？？？
引数が可変（フォームの必須/非必須により）で数も多い場合のアプローチ

- 入力項目数分の引数を書いて受け取る
```
fun recieve_input (
    a: u64,
    b: u64,
    c: String,
    d: u32,
    ...
    ctx: &mut TxContext
)
```

- ひとまとめにして受け取り、関数内で処理する
    - この場合どのようにしたら良いのか？？？
```
fun recieve_input (
    form_input,
    ctx: &mut TxContext
)
```

**Comment by 0x-onigiri** [comment]
自分の学習した範囲内では、

> ひとまとめにして受け取り、関数内で処理する

これはオブジェクトにしないと難しいと思っています

なので、

> 入力項目数分の引数を書いて受け取る

の方法しか思いつかなかったです〜

**Reply by fukaeridesui** [reply]
ありがとうございます！
やっぱまとめては難しそうなんですね、、引数が多い関数ができそうです😅
